tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -2558,7 +2566,11 @@ HRESULT CorProfiler::CallTarget_RewriterCallback(RejitHandlerModule* moduleHandl
     bool isVoid = (retTypeFlags & TypeFlagVoid) > 0;
     bool isStatic = !(caller->method_signature.CallingConvention() & IMAGE_CEE_CS_CALLCONV_HASTHIS);
     std::vector<FunctionMethodArgument> methodArguments = caller->method_signature.GetMethodArguments();
-    int numArgs = caller->method_signature.NumberOfArguments();
+    bool useCustomArgumentsTargetMethodArguments = integration_definition->target_method.use_target_method_arguments_to_load;
+    std::vector<USHORT> methodArgumentsToLoad = integration_definition->target_method.target_method_arguments_to_load;
+    int numArgs = useCustomArgumentsTargetMethodArguments
+                    ? (int) methodArgumentsToLoad.size()
+                    : caller->method_signature.NumberOfArguments();
```

Comment:
Just speculating, but do we expect any performance impact (hopefully improvement) from not passing all the arguments to our integrations? e.g. should we adopt this as "standard" for our integrations, where we only pass the arguments we strictly need?

Reply:
I'm not sure how much we would save from avoiding unnecessary `ldarg` IL instructions, but maybe it could reduce the amount of generic types are created at runtime? This would be good to benchmark at some point